src/main/java/io/anserini/collection/ClueWeb09.java

```diff
@@ -0,0 +1,745 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Following header retained per conditions specified in license.
+
+/*
+ * Container for a generic Warc Record
+ *
+ * (C) 2009 - Carnegie Mellon University
+ *
+ * 1. Redistributions of this source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. The names "Lemur", "Indri", "University of Massachusetts",
+ *    "Carnegie Mellon", and "lemurproject" must not be used to
+ *    endorse or promote products derived from this software without
+ *    prior written permission. To obtain permission, contact
+ *    license@lemurproject.org.
+ *
+ * 4. Products derived from this software may not be called "Lemur" or "Indri"
+ *    nor may "Lemur" or "Indri" appear in their names without prior written
+ *    permission of The Lemur Project. To obtain permission,
+ *    contact license@lemurproject.org.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE LEMUR PROJECT AS PART OF THE CLUEWEB09
+ * PROJECT AND OTHER CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN
+ * NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author mhoy@cs.cmu.edu (Mark J. Hoy)
+ */
+
+package io.anserini.collection;
+
+import io.anserini.document.SourceDocument;
+
+import java.io.BufferedReader;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Class representing an instance of the
+ * <a href="https://www.lemurproject.org/clueweb09.php/">ClueWeb09 collection</a>.
+ * This can be used to read the complete ClueWeb09 collection or the smaller ClueWeb09b subset.
+ */
+public class ClueWeb09 extends Collection {
```

Comment:
I assume we will be changing other collection names too?

Reply:
Yes, if you like this, I will slowly change other collection names as we refactor. `ClueWeb09Collection` sounds redundant, but I'd keep `TrecCollection`?
Or would you rather everything be named `FooCollection`?

Reply:
Yea, I am thinking this change will break everyone's local scripts... which is bad...
I prefer FooCollection since it is pretty recognizable and keep it as is makes everyone happy